nucypher/characters/control/interfaces.py

```diff
@@ -163,27 +161,14 @@ class BobInterface(CharacterPublicInterface):
 
     @attach_schema(bob.RetrieveAndDecrypt)
     def retrieve_and_decrypt(self,
-                             policy_encrypting_key: bytes,
-                             alice_verifying_key: bytes,
-                             message_kit: bytes,
-                             encrypted_treasure_map: Union[bytes, str, 'EncryptedTreasureMap']):
+                             policy_encrypting_key: PublicKey,
+                             alice_verifying_key: PublicKey,
+                             message_kits: List[MessageKit],
+                             encrypted_treasure_map: EncryptedTreasureMap) -> dict:
         """
         Character control endpoint for re-encrypting and decrypting policy data.
         """
-        from nucypher.characters.lawful import Enrico
-
-        policy_encrypting_key = PublicKey.from_bytes(policy_encrypting_key)
-        alice_verifying_key = PublicKey.from_bytes(alice_verifying_key)
-        message_kit = MessageKit.from_bytes(message_kit)  # TODO #846: May raise UnknownOpenSSLError and InvalidTag.
-
-        if isinstance(encrypted_treasure_map, bytes):
```

Comment:
I suppose the thinking here is that deserialization must be handled by the caller?

Reply:
The EncryptedTreasureMap field used in the RetrieveAndDecrypt Schema, automatically handles the deserialization - https://github.com/derekpierre/nucypher/blob/porter-retrieval/nucypher/characters/control/specifications/fields/treasuremap.py#L27.